crates/dojo-defi/src/constant_product_market/systems.cairo

```diff
@@ -88,4 +91,102 @@ mod Sell {
+
+        // update market
+        commands::set_entity(
+            market_sk,
+            (Market {
+                cash_amount: market.cash_amount + cost_cash,
+                item_quantity: market.item_quantity + cost_quantity
+            })
+        );
+
+        // update player cash
+        commands::set_entity(cash_sk, (Cash { amount: player_cash.amount - cost_cash }));
+
+        // update player item
+        commands::set_entity(item_sk, (Item { quantity: player_quantity - cost_quantity }));
+    // update player liquidity
+    // TODO: uncomment once error: Plugin diagnostic: Type not found is solved
+    // let liquidity_sk: Query = (partition, (player, item_id)).into_partitioned();
+    // let player_liquidity = commands::<Liquidity>::entity(liquidity_sk);
+    // commands::set_entity(
+    //     lquidity_sk, (Liquidity { shares: player_liquidity.shares + liquidity_shares })
+    // );
+    }
+}
+// TODO: uncomment once error: Plugin diagnostic: Type not found is solved
+// #[system]
+// mod RemoveLiquidity {
+//     use traits::Into;
+//     use array::ArrayTrait;
+//     use dojo_core::integer::u250;
+//     use dojo_core::integer::ContractAddressIntoU250;
+//     use dojo_defi::constant_product_market::components::Item;
+//     use dojo_defi::constant_product_market::components::Cash;
+//     use dojo_defi::constant_product_market::components::Market;
+//     use dojo_defi::constant_product_market::components::MarketTrait;
+
+//     fn execute(partition: u250, item_id: u250, shares: FixedType) {
+//         let player: u250 = starknet::get_caller_address().into();
+
+//         let liquidity_sk: Query = (partition, (player, item_id)).into_partitioned();
+//         let player_liquidity = commands::<Liquidity>::entity(liquidity_sk);
+//         assert(player_liquidity.shares >= shares, 'not enough shares');
+
+//         let market_sk: Query = (partition, (item_id)).into_partitioned();
+//         let market = commands::<Market>::entity(market_sk);
+//         let (payout_cash, payout_quantity) = market.remove_liquidity(shares);
+
+//         // update market
+//         commands::set_entity(
+//             market_sk,
+//             (Market {
+//                 cash_amount: market.cash_amount + cost_cash,
+//                 item_quantity: market.item_quantity + cost_quantity
+//             })
+//         );
+
+//         // update player cash
+//         commands::set_entity(cash_sk, (Cash { amount: player_cash.amount + cost_cash }));
+
+//         // update player item
+//         commands::set_entity(item_sk, (Item { quantity: player_quantity + cost_quantity }));
+//         // update player liquidity
+//         let liquidity_sk: Query = (partition, (player, item_id)).into_partitioned();
+//         let player_liquidity = commands::<Liquidity>::entity(liquidity_sk);
+//         commands::set_entity(lquidity_sk, (Liquidity { shares: player_liquidity.shares - shares }));
+//     }
+// }
+
```

Comment:
For now, I've commented this out because I'm getting `error: Plugin diagnostic: Type not found.`, will fix this once I get more info on how to solve it

Reply:
dont you need to import FixedType and Liquidity ? same as you do for Market for example

Reply:
You're right, I was importing it on the `Buy` system and not in `AddLiquidity` 🤦 . Now it seems to work, however for `RemoveLiquidity` I'm getting this error, would you happen to know why it's throwing this?

```
error: Trait has no implementation in context: core::serde::Serde::<cubit::types::fixed::FixedType>
 --> contract:221:44
                serde::Serde::<FixedType>::deserialize(ref data).expect('Input too short for arguments');
                                           ^*********^

Error: failed to compile
```

Reply:
I managed to fix it by implementing Serde through `#[derive]` attribute, created a pr for it https://github.com/influenceth/cubit/pull/5. For now I'll use my fork and add a todo